pkg/bzz/address.go

```diff
@@ -28,10 +28,11 @@ var ErrInvalidAddress = errors.New("invalid address")
 // It consists of a peers underlay (physical) address, overlay (topology) address and signature.
 // Signature is used to verify the `Overlay/Underlay` pair, as it is based on `underlay|networkID`, signed with the public key of Overlay address
 type Address struct {
-	Underlay    ma.Multiaddr
-	Overlay     swarm.Address
-	Signature   []byte
-	Transaction []byte
+	Underlay        ma.Multiaddr
+	Overlay         swarm.Address
+	Signature       []byte
+	Transaction     []byte
+	EthereumAddress []byte
```

Comment:
Why isn't it sufficient to put into p2p.Peer? Why add it to the addressbook?

Reply:
how will you recover it from the outside? recovering requires the `generateSignData` logic which is only available here. It could be exported, then we could do the same computation another time somewhere else.

Reply:
Isn't this PR about preparing the removal of beneficiary check we do in the swap protocol because the handshake should now pass the public key?